### ISSUE TITLE
Controller mapping popup is ferociously annoying

### ISSUE THREAD

**Issue by commie-caden** [xbmc/xbmc#23777]
### Describe the bug

Whenever I go to configure a controller's button mappings, I must face the dreaded popups. Every single time I get to either the d-pad, triggers, or analog sticks, I get a helpful reminder to move the analog sticks in the direction seen in the list. It's entirely useless and very frustrating. Those items are already clear and easy to understand, this popup actually prevents me from entering the inputs, and is a huge pain without a permanent way to disable it. Please, just remove it completely!

## Expected Behavior

I just want to map my controller into Kodi.

## Actual Behavior

I need to speed run the input mapping to hopefully beat the popup that appears moments after clicking enter to map the triggers or analog sticks or d-pad.

## Possible Fix

It should just get removed entirely.

### Screenshots

![Screenshot from 2023-09-17 03-37-43](https://github.com/xbmc/xbmc/assets/25536957/0d6ed42f-e3d2-4b19-8b86-12b76407a4ef)

### Your Environment
Used Operating system:
 - [ ] Android
 - [ ] iOS
 - [ ] tvOS
 - [x] Linux
 - [ ] OSX
 - [ ] Windows
 - [ ] Windows UWP

 - Operating system version/name: Fedora Linux 38 (Workstation Edition), Libreelec
 - Kodi version: 20.2 (Flatpak)

**Comment by garbear** [comment]
I agree, the popup annoys even me, and I created it 🙂 However, we can't just take down Chesterton's Fence. The joystick drivers for your OS don't behave very well, and we can't know that an analog button, trigger or analog stick exists until it has been activated.

You'll notice that, on first Kodi boot, if you press every button, then when you go to map the controller you won't get the popup, because Kodi already knows about every button and doesn't see them for the first time. But this isn't the best solution to hiding the popup because when Kodi is restarted, it forgets which analog buttons actually exists.

A way for Kodi to remember analog buttons across restarts would fix the popup being shown when Kodi is restarted. You would still see the popup the first time you install Kodi, but after that never again.

When I get a chance to work on Kodi next, I'll prioritize this "remembering" so we can hide the dialog and speedrun our controller mapping.

**Comment by TriVoxel** [comment]
This is my experience: I am on Fedora Linux which uses the standard controller libraries and whatnot. All the controllers I use are supported by SDL, which is the system controller library. This library already contains the button information for most controllers. Likely, you could just automatically assume controller bindings based on SDL's defaults. SLD tends to have very similar conventions across controllers, ie. `[button 0]` is primary/A, `[button 1]` is secondary/B, etc. You could reasonably assume a controller's config by virtue of all SDL controllers using similar configs. The user could then override this in input settings.

My problem is simply this popup, though. I only encounter it once, when setting up a new model of controller for the first time, and only in the settings. Never have an issue on subsequent launches. I've battled this popup once for my Wii U controller, once for my Switch, once for my Xbox Series, etc. It's really annoying, though, because in the controller setup wizard, each input asks this stupid prompt. I'm okay with binding a controller manually, but having this popup regardless of whether I'm entering inputs, is really annoying. I mean, if you want to inform the user that they need to wiggle the analog sticks, maybe instead of a popup just change the text in the text box below when they are on the analog stick section? OR, if keeping this popup is necessary, maybe only show it if the user gets to the analog binding phase and isn't sending inputs? Currently, there seems to only be a timer which goes off regardless of whether the user is doing what they're supposed to, which actually just impedes what the popup is trying to inform them of, completely defeating the purpose.

Again, I honestly, just don't think this popup is necessary considering the average Kodi user would need to be somewhat technically literate, and likely already understands that they need to enter analog inputs when they get to the joystick section. This popup is likely completely redundent, and only inconveniences the user. My vote is just to have it completely removed, or for the text box to change descriptions depending on which button is being bount in the setup process in settings, system, input, configure attached devices.

**Comment by garbear** [comment]
I did some digging and the dialog was initially introduced here: https://github.com/xbmc/xbmc/pull/11623

You can see I outlined the problem description, how I solved it using this dialog, and regrets over how annoying it is. Finally, I ask for suggestions on a better solution than the ferociously annoying dialog.

As I was then, I'm more than open to alternative solutions.

**Comment by TriVoxel** [comment]
Couldn’t you read controller input during this popup? That would sorta fix the whole problem as well. The input mapper stops reading input until the popup is dismissed. Or, you could out the message in-line with the mapping wizard.

**Comment by garbear** [comment]
Controller input _is_ read during the popup. The problem is that there is no input _until_ the popup. The driver only reports changes, so a centered thumbstick would show up for the first time when mapping. And if the first motion pushes the stick fully down and the first value is -1.0, it can't be distinguished from an anomalous trigger, which travels from -1.0 to 1.0.

One solution would be to remember the axes by writing some config to XML. The dialog is still needed, but only when wiping userdata, not when restarting Kodi. Showing the dialog once instead of after every restart would be much less annoying.

**Comment by garbear** [comment]
Annoyingness is no less ferocious today.
